docs/compass/10-01-rate-limiting.md

```diff
@@ -0,0 +1,28 @@
+
+## EnvoyFilters in CMP 
+
+We have created 2 rate-limiting *EnvoyFilter* resources:
+
+- *EnvoyFilter* for **Compass Gateway**
+- *EnvoyFilter* for **Compass ORD-Service**
+
+Since most of our auth flows pass through the Gateway component (the upstream for the majority of our [ORY rules](../../chart/compass/charts/gateway/templates/oathkeeper-rules.yaml) is the gateway service) it is an appropriate service to apply rate-limiting which captures most of the incoming requests to CMP. Several flows end up in the ord-service and that is why the second *EnvoyFilter* is applied there. The rest of the flows are either for internal calls (no rate-limiting needed) or for the system-broker component for which CMP does not apply rate-limiting.
+ 
+### Gateway EnvoyFilter
+
+The [Gateway EnvoyFilter](../../chart/compass/charts/gateway/templates/rate-limit-filter.yaml) leverages [descriptors](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#using-rate-limit-descriptors-for-local-rate-limiting) and thus describes 3 token buckets. A *descriptor* is a key-value pair which provides information about the current request. *Descriptors* are created by *actions*. 
+
+The gateway EnvoyFilter has 1 action which creates a descriptor with a key `consumer`. The value of that descriptor for each HTTP request is retrieved from an HTTP header which in the default configuration is `X-Flow-Identity`. This header is added in the **compass-director-certs** and **compass-connector-certs** auth flows (described by the [ORY rules](../../chart/compass/charts/gateway/templates/oathkeeper-rules.yaml) with the same names) by an ORY Header Hydrator which inspects the ORY session to get the value for it. The value of the header is the type of the consumer (*Application*, *Runtime*  or *Integration System*). Thus, the `consumer` descriptor's value can be one of those 3 possible consumer types. 
+
+In case the value of the descriptor is *Runtime* or *Application* the request will get matched by the EnvoyFilter and a separate token-bucket will be used to determine whether to rate-limit the request or not. Otherwise, the *general* token-bucket will be used. Notice that all other auth flows (except from **compass-director-certs** and **compass-connector-certs**) match [ORY rules](../../chart/compass/charts/gateway/templates/oathkeeper-rules.yaml) which don't add the `X-Flow-Identity` header, respectively the `consumer` descriptor is not created for them meaning that they also end up in the *general* token-bucket. The following diagram depicts the flow:
+
+![](./assets/gateway-rate-limiting.svg)
```

Comment:
Add white background to the diagram itself, otherwise, it is not readable on dark theme of github